app/controllers/api/v1/downloads_controller.rb

```diff
@@ -44,6 +44,26 @@ def batch_export
         download(storage, file_location)
       end
 
+      # GET /downloads/batch_import/:id(/:option)
+      def batch_import
+        batch_import = load_and_authorize_batch_import!
+
+        if batch_import_params[:option] == 'without_successful_imports'
+          send_data(
```

Comment:
Most of the time, this `send_data` call will be sending a relatively small file, but sometimes it might be a much larger CSV with a lot of columns.  It would be good to refactor this to stream like the other downloads methods, using DOWNLOAD_BUFFER_BYTE_SIZE to limit the amount of data in memory at any given moment.  This is a sort of tricky situation though, given how we need to read and filter the csv without successful rows before providing it as a download.

Here here's one possible option:
- Convert `csv_without_successful_imports` to a method that does a buffered low-memory-usage write to a temporary file on disk, and have it yield a block with a reference to that file, and automatically clean up the temporary file afterward in an `ensure`.
- In the block that you run, read the temporary file and stream it via the same `response.stream.write(chunk)` strategy used in the `download` method.  Ideally it would be nice to re-use more of the logic in the download method, and maybe we can with some additional refactoring, but the above steps would be a first-pass way of keeping the memory usage low for this download case.

Reply:
We currently use `send_data` in prod and we haven't seen a problem yet (that I know of). Refactoring this can be delayed. I think we should just ticket this and work on it later when we have time.

Reply:
Sounds good! Ticketed!